File: backend/micro_epoweredmove_plug_plugtype/src/main/java/com/kosdiam/epoweredmove/models/Plug.java
package com.kosdiam.epoweredmove.models;

import java.io.Serializable;

import com.kosdiam.epoweredmove.models.enums.Availability;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Plug implements Serializable {
    private String id;
    private Availability availability;
    private Float power;
    private Integer phases;
    private String chargingStationId;
    private Long timestamp;
    private String plugTypeId;

}

File: frontend/E-PoweredMoveAndroidApp/app/src/main/java/com/kosdiam/epoweredmove/models/Plug.kt
package com.kosdiam.epoweredmove.models

import android.net.Uri
import android.os.Parcelable
import com.kosdiam.epoweredmove.models.enums.Availability
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Plug(
    val id: String,
    val availability: Availability,
    val power: Float,
    val phases: Int,
    val chargingStationId: String,
    val chargingStationObj: ChargingStation,
    val timestamp: Long,
    val plugTypeId: String,
    val plugTypeObj: PlugType
) : Parcelable{
    constructor(): this("", Availability.AVAILABLE, 0f, 0, "", ChargingStation(), 0, "", PlugType())
}
